lib/Mojo/Loader.pm

```diff
@@ -32,11 +32,31 @@ sub find_packages {
   return sort map { /^(.+)::$/ ? "${ns}::$1" : () } keys %{"${ns}::"};
 }
 
+sub is_package {
+  my $package = shift;
+  if($package && $package =~
+    /
+      ^
+      [a-zA-Z_]         # Can't begin from digit
+      [a-zA-Z0-9_]*
+      (?:
+          (?: :: | ' )  # Package::name eq Package'name
+          [a-zA-Z0-9_]+
+      )*
+      \z
+    /x) {
+      return 1;
+  }
+  return 0;
+}
```

Comment:
That probably fits on one line.

Reply:
You meant
$package && $package =~ /.../ ? 1 : 0;
or collapse of regexp.

Reply:
All.

Reply:
If you don't want to emulate the Mojolicious coding style then this code does not belong in Mojolicious. That regex is not rocket science. We only expand non-trivial regexes with comments.

Reply:
But aside from that, i'm more concerned with the fact that this is a custom regex. It should really have been an existing one copied from a proven module.